src/SdfEntityCreator.cc

```diff
@@ -220,13 +220,29 @@ Entity SdfEntityCreator::CreateEntities(const sdf::World *_world)
 {
   GZ_PROFILE("SdfEntityCreator::CreateEntities(sdf::World)");
 
-  // World entity
-  Entity worldEntity = this->dataPtr->ecm->CreateEntity();
+  Entity worldEntity = kNullEntity;
+  this->dataPtr->ecm->Each<gz::sim::components::Name,
+            gz::sim::components::World>(
+    [&](const gz::sim::Entity &_entity,
+        const gz::sim::components::Name *_name,
+        const gz::sim::components::World *)->bool
+  {
+    if (_world->Name() == _name->Data())
+    {
+      worldEntity = _entity;
+      return false;
```

Comment:
Should the return be `true` here and `false` below?

Reply:
If `false` the `Each` loop breaks, isn't it? then it's fine like this.

Reply:
Ah yes, you are correct.